platformdirs.py

```diff
@@ -277,6 +279,78 @@ def _user_log_dir_impl(appname=None, appauthor=None, version=None, opinion=True)
 
         return path
 
+elif system == 'android':
+
+    def _get_android_folder_with_pyjnius():
+        from jnius import autoclass
+
+        Context = autoclass('android.content.Context')
+        return Context.getFilesDir().getParentFile().getAbsolutePath()
+
+    def _get_android_folder_brute_path():
+        from re import match
+
+        pattern = r'/data/(data|user/\d+)/(.+)/files'
+
+        for path in sys.path:
+            if match(pattern, path):
+                return path.split('/files')[0]
+
+        raise OSError('Cannot find path to android app folder')
+
+    try:
+        _get_android_folder_with_pyjnius()
+    except:
+        _get_android_folder = _get_android_folder_brute_path
+    else:
+        _get_android_folder = _get_android_folder_with_pyjnius
```

Comment:
Should we add a `finally` where we cache the result of `_get_android_folder`?

Reply:
yes, I will correct it